source/ingress_test.go

```diff
@@ -1176,6 +1191,57 @@ func testIngressEndpoints(t *testing.T) {
 			},
 		},
 		{
+			title:                  "ingressClassName filtering",
+			targetNamespace:        "",
+			ingressClassNames:      []string{"public", "dmz"},
+			ingressItems: []fakeIngress{
+				{
+					name:             "fake-public",
+					namespace:        namespace,
+					tlsdnsnames:      [][]string{{"example.org"}},
+					ips:              []string{"1.2.3.4"},
+					ingressClassName: "public",
+				},
+				{
+					name:             "fake-internal",
+					namespace:        namespace,
+					tlsdnsnames:      [][]string{{"int.example.org"}},
+					ips:              []string{"2.3.4.5"},
+					ingressClassName: "internal",
+				},
+				{
+					name:             "fake-dmz",
+					namespace:        namespace,
+					tlsdnsnames:      [][]string{{"dmz.example.org"}},
+					ips:              []string{"3.4.5.6"},
+					ingressClassName: "dmz",
+				},
+				{
+					name:             "annotated-dmz",
+					namespace:        namespace,
+					tlsdnsnames:      [][]string{{"annodmz.example.org"}},
+					ips:              []string{"4.5.6.7"},
+					annotations: map[string]string{
+						"kubernetes.io/ingress.class": "dmz",
+					},
+				},
```

Comment:
Suggest test case with annotation dmz and ingressClassName internal. Should not be in expected.

Possibly also annotation internal ingressClassName dmz. Should be in expected.

Suggest test case with neither annotation nor ingressClassName. Should not be in expected.